migration-engine/migration-engine-tests/tests/migrations/indexes.rs

```diff
@@ -1035,3 +1035,100 @@ fn do_not_overwrite_fulltext_index_without_preview_feature(api: TestApi) {
 
     api.schema_push_w_datasource(dm).send().assert_no_steps();
 }
+
+#[test_connector(tags(Mysql), preview_features("fullTextIndex"))]
+fn adding_fulltext_index_to_an_existing_column(api: TestApi) {
+    let dm = formatdoc! {r#"
+        {}
+
+        generator client {{
+          provider = "prisma-client-js"
+          previewFeatures = ["fullTextIndex"]
+        }}
+
+        model A {{
+          id Int    @id
+          a  String @db.Text
+          b  String @db.Text
+        }}
+    "#, api.datasource_block()};
+
+    api.schema_push(&dm).send().assert_green();
+
+    api.assert_schema()
+        .assert_table("A", |table| table.assert_indexes_count(0));
+
+    let dm = formatdoc! {r#"
+        {}
+
+        generator client {{
+          provider = "prisma-client-js"
+          previewFeatures = ["fullTextIndex"]
+        }}
+
+        model A {{
+          id Int    @id
+          a  String @db.Text
+          b  String @db.Text
+
+          @@fulltext([a, b])
+        }}
+    "#, api.datasource_block()};
+
+    api.schema_push(&dm).send().assert_green();
+
+    api.assert_schema().assert_table("A", |table| {
+        table.assert_index_on_columns(&["a", "b"], |index| index.assert_is_fulltext())
+    });
+}
+
+#[test_connector(tags(Mysql), preview_features("fullTextIndex"))]
+fn changing_fulltext_index_to_a_normal_index(api: TestApi) {
```

Comment:
nitpick: i think this one should be called `changing_normal_index_to_a_fulltext_index`